From 71776e673e7a75f2b4c9b13a618baaba278ca295 Mon Sep 17 00:00:00 2001
From: Jean Trivelly <jean.trivelly@intel.com>
Date: Wed, 14 Aug 2013 11:59:04 +0200
Subject: [PORT FROM MAIN] wifi: update UI connection status as soons as association step starts
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

BZ: 130799 123533

This patch makes wpa_supplicant transitions to state “WPA_ASSOCIATING” after
"current_ssid" gets updated.
Otherwise upper layer (Android and thus UI) are not warned a connection is
ongoing before authentication step.

By doing so, upper layers and UI status are set to “Connecting” as soon
as the association phase starts.

Change-Id: I007d910b963f2b08a1471c66b0a2daf049b77f31
Orig-Change-Id: I86585f949885771ce57a61d24f6014df0ae16bee
Category: device enablement
Domain: CWS.WIFI-Common
Origin: upstream
Upstream-Candidate: yes
Signed-off-by: Jeremie GARCIA <jeremie.garcia@intel.com>
---
 wpa_supplicant/wpa_supplicant.c |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/wpa_supplicant/wpa_supplicant.c b/wpa_supplicant/wpa_supplicant.c
index 0b5bc34..18b78f7 100644
--- a/wpa_supplicant/wpa_supplicant.c
+++ b/wpa_supplicant/wpa_supplicant.c
@@ -1544,7 +1544,9 @@ void wpa_supplicant_associate(struct wpa_supplicant *wpa_s,
 		wpa_supplicant_set_wpa_none_key(wpa_s, ssid);
 	}
 
+#ifndef ANDROID
 	wpa_supplicant_set_state(wpa_s, WPA_ASSOCIATING);
+#endif
 	if (bss) {
 		params.ssid = bss->ssid;
 		params.ssid_len = bss->ssid_len;
@@ -1709,6 +1711,13 @@ void wpa_supplicant_associate(struct wpa_supplicant *wpa_s,
 	}
 	old_ssid = wpa_s->current_ssid;
 	wpa_s->current_ssid = ssid;
+#ifdef ANDROID
+	/*
+	 * Update state except if already completed (open network)
+	 */
+	if (wpa_s->wpa_state != WPA_COMPLETED)
+		wpa_supplicant_set_state(wpa_s, WPA_ASSOCIATING);
+#endif
 	wpa_s->current_bss = bss;
 	wpa_supplicant_rsn_supp_set_config(wpa_s, wpa_s->current_ssid);
 	wpa_supplicant_initiate_eapol(wpa_s);
-- 
1.7.4.1

